cpp/ql/test/query-tests/Best Practices/Likely Errors/CommaBeforeMisleadingIndentation/test.cpp

```diff
@@ -77,6 +77,10 @@ int test(int i, int j, int (*foo)(int), int (*bar)(int, int))
 	                   j++);
 	(void)sizeof(x.foo(i++), // GOOD
 	             j++);
+	using U = decltype(x.foo(i++), // GOOD? Unlikely to be misread
+				j++);
+	(void)sizeof(x.foo(i++), // GOOD? Unlikely to be misread
+				j++);
```

Comment:
Ouch. I feel bad for whoever ends up reading code like this. This seems like an obvious case of https://wiki.sei.cmu.edu/confluence/display/cplusplus/EXP52-CPP.+Do+not+rely+on+side+effects+in+unevaluated+operands.

Reply:
Ouch indeed. There is a certain popular header-only JSON library that does [something like that](https://github.com/syoyo/tinygltf/blob/9bb5806df4055ac973b970ba5b3e27ce27d98148/json.hpp#L4188-L4188). Kept popping up in the MRVA top 1000.